common.c

```diff
@@ -62,8 +62,50 @@ const Instr_t Primes[PROGRAM_SIZE] = {
     Instr_Halt           // nmax, c (== nmax)
 };
 
+/* Program for counting square root in integer numbers
+ * with Newton-Raphson method */
+const Instr_t Newton[PROGRAM_SIZE] = {
+    Instr_Push, 0,      // d (decreased = false)             0
+    Instr_Push, 9999,   // d, n (we are looking for sqrt(n)) 2
+    Instr_Push, 1,      // d, n, x (x = sqrt(n))             4
+    /* loop: */
+    Instr_Over,         // d, n, x, | n                      6
+    Instr_Over,         // d, n, x, | n, x                   7
+    Instr_Swap,         // d, n, x, | x, n                   8
+    Instr_Div,          // d, n, x, | n/x                    9
+    Instr_Over,         // d, n, x, | n/x, x                 10
+    Instr_Add,          // d, n, x, | x+n/x                  11
+    Instr_Push, 2,      // d, n, x, | x+n/x, 2               12
+    Instr_Swap,         // d, n, x, | 2, x+n/x               14
+    Instr_Div,          // d, n, x, | (x+n/x)/2 = nx         15
+    Instr_Over,         // d, n, x, | nx, x                  16
+    Instr_Over,         // d, n, x, | nx, x, nx              17
+    Instr_Sub,          // d, n, x, | nx, nx-x               18
+    Instr_JE, +21, /* end */ // d, n, x, | nx                19
+    Instr_Over,         // d, n, x, | nx, x                  21
+    Instr_Over,         // d, n, x, | nx, x, nx              22
+    Instr_Greater,      // d, n, x, | nx, nx>x               23
+    Instr_JNE, +9, /* stop */ // d, n, x, | nx               24
+    Instr_Push, 1,      // d, n, x, | nx, 1                  26
+    Instr_Push, 0,      // d, n, x, | nx, 1, 0               28
+    Instr_Set,          // 1, n, x, | nx                     30
+    /* continue */
+    Instr_Swap,         // d, n, nx, x                       31
+    Instr_Drop,         // d, n, nx                          32
+    Instr_Jump, -29, /* loop */ // d, n, nx                  33
+    /* stop */
+    Instr_Push, 0,      // d, n, x, | nx, 0                  35
+    Instr_Get,          // d, n, x, | nx, d                  37
+    Instr_JNE, +2, /* end */ // d, n, x, | nx                38
+    Instr_Jump, -11, /* continue */     // d, n, x, | nx     40
+    /* end */
+    Instr_Drop,         // d, n, x                           42
+    Instr_Print,        // d, n                              43
+    Instr_Halt          //                                   44
+};
+
 /* Choose a program we are about to simulate */
-const Instr_t* Program = Primes;
+const Instr_t* Program = Newton;
```

Comment:
Will be changed back to `Primes` for default build, but for tests, I will use `Newton` of course.